lib/dfp_helper.rb

```diff
@@ -39,13 +39,13 @@ def dfp_define_slot(_i, options = {})
 </script>
       END
     end
-    
+
     def dfp_helper_head(options={single_request:true})
       return unless dfp_helper_slots.size > 0
       o = dfp_helper_slots.collect{|i|
         _targeting = (i[:targeting]||[]).collect{|k,v| ".setTargeting(#{k.to_json}, #{v.to_json})"}.join
         _slot_name = (i[:slot_name].blank?)?"":"#{i[:slot_name]} = "
-        "#{_slot_name}googletag.defineSlot('#{i[:id]}', [#{i[:size].map(&:to_s).join(', ')}], '#{i[:div_id]}').addService(googletag.pubads())#{_targeting};"
+        "#{_slot_name}googletag.defineSlot('#{i[:id]}', [#{i[:size].map(&:to_s).join(', ')}], '#{i[:div_id]}').setCollapseEmptyDiv(true).addService(googletag.pubads())#{_targeting};"
```

Comment:
the only other way if GPT doesn't expose a way of getting a slot object would be to store a registry of slots ourselves
```
slots[#{slot_name}] = #{_slot_name}googletag.defineSlot ...
```